doc/samples/why_new_template.rst

```diff
@@ -56,6 +59,60 @@ All below commands assumes you are in :file:`{NCS_PATH}/sidewalk/samples` and bu
+| Description                    | Old                      | New                     |
++================================+==========================+=========================+
+| Send Hello                     | ``Button 3 (short)``     | ``Button 1 (short)``    |
++--------------------------------+--------------------------+-------------------------+
+| Enter DFU mode                 | ``Button4 (long)``       | ``Button 1 (long)``     |
++--------------------------------+--------------------------+-------------------------+
+| Factory Reset                  | ``Button1 (long)``       | ``Button 2 (long)``     |
++--------------------------------+--------------------------+-------------------------+
+| Board support                  | ``board_events.c``       | N/A (use CLI command)   |
++--------------------------------+--------------------------+-------------------------+
+| Set fake battery level         | ``Button 4 (short)``     | N/A (use CLI command)   |
++--------------------------------+--------------------------+-------------------------+
+| Get current Device Profiles    | ``Button 2 (short)``     | N/A (use CLI command)   |
++--------------------------------+--------------------------+-------------------------+
+| Switch between Device Profiles | ``Button 2 (long)``      | N/A (use CLI command)   |
++--------------------------------+--------------------------+-------------------------+
+| Sensor Monitoring button press | ``Buttons 1-4 (short)``  | ``Buttons 1-4 (short)`` |
++--------------------------------+--------------------------+-------------------------+
+| Switch Link Mask               | N/A                      | ``Buttons 3 (long)``    |
++--------------------------------+--------------------------+-------------------------+
+
+
+**LEDs** changes:
+
++--------------------------------+--------------+--------------+
+| Description                    | Old          | New          |
++================================+==============+==============+
+| Application Link and Connected | ``LED 1``,   | ``LED 1``    |
+|                                | ``LED 4``    |              |
++--------------------------------+--------------+--------------+
+| Application time sync          | ``LED 3``    | ``LED 2``    |
++--------------------------------+--------------+--------------+
+| Application Registered         | ``LED 2``    | ``LED 3``    |
++--------------------------------+--------------+--------------+
+| Application woke up            | N/A          | ``LED 4``    |
++--------------------------------+--------------+--------------+
+| Sensor Monitoring led          | ``LEDs 1-4`` | ``LEDs 1-4`` |
++--------------------------------+--------------+--------------+
+
+
+**CLI** changes:
+
+* All Sidewalk Device Under Test application (``sid_dut``) commands are available when ``CONFIG_TEMPLATE_APP_CLI`` is enabled.
+* Old ``CONFIG_SIDEWALK_CLI`` was removed.
+
+
```

Comment:
@ktaborowski In the reviewed migration guide, that just got merge, in the "source files" section I left "samples" in the description field but it's not correct. However, I couldn't come up with a term that would encapsulate all of the terms listed - could you please take a look at that description as well? Thank you!